File: golden-paths/java/spring-boot/src/test/java/lab/weyland/golden/selfcheck/SelfCheckTest.java
package lab.weyland.golden.selfcheck;

import static org.junit.jupiter.api.Assertions.assertEquals;

import org.junit.jupiter.api.Test;

/**
 * Deliberately-failing test — proves the java lane PROPAGATES failure (B88 --self-check runs the
 * `selfcheck` profile, which includes ONLY this). Fails on a real assertion, not a bare fail.
 */
class SelfCheckTest {
    @Test
    void theLaneReportsFailure() {
        assertEquals("hello, weyland", "this fixture is meant to fail",
                "selfcheck: the golden-path java lane must surface this failure (exit non-zero)");
    }
}

File: golden-paths/dart/shelf/test/selfcheck_test.dart
// Deliberately-failing test tagged `selfcheck` — proves the dart lane PROPAGATES failure.
// A normal `dart test` skips it (dart_test.yaml → tags.selfcheck.skip → exit 0); the lane
// self-check runs `dart test -t selfcheck --run-skipped`, which forces the tagged test to
// actually execute and exit non-zero. See dart_test.yaml for why --run-skipped is required.
@Tags(['selfcheck'])
library;

import 'package:test/test.dart';

void main() {
  test('selfcheck: the golden-path dart lane must surface this failure', () {
    fail('selfcheck: the golden-path dart lane must surface this failure (exit non-zero)');
  });
}
